### ISSUE TITLE
Add Gradle

### ISSUE THREAD

**Issue by vorburger** [quarkusio/quarkus-quickstarts#75]
started in https://github.com/quarkusio/quarkus-quickstarts/compare/master...vorburger:gradle

currently blocked by https://github.com/quarkusio/quarkus/issues/1323

@stalep FYI

**Comment by vorburger** [comment]
quarkusio/quarkus#1323 is now fixed, and when there is a new Quarkus 0.12.0 release I'll bump the versions in getting-started/build.gradle of master...vorburger:gradle and raise a PR from that branch.

**Comment by vorburger** [comment]
@jnizet given your https://github.com/quarkusio/quarkus/pull/1403, shout via comments here if you have an feedback on https://github.com/quarkusio/quarkus-quickstarts/compare/master...vorburger:gradle ... (no PR actually raised yet, as waiting for release, or if you prefer I could raise a DRAFT PR that you can edit?).

**Comment by jnizet** [comment]
@vorburger My feedback would be to rey to avoid `apply plugin xxx`, and instead do as I suggest in https://github.com/quarkusio/quarkus/pull/1403.

Applying plugins imperatively that way is discouraged for a long time by Gradle in favor of using the plugins block. 

The main reasons are that
- it's simple (at least when the plugin is deployed in the gradle plugin central, otherwise you need to adapt the settings.gradle file), because you don't have to add anything to the buildscript classpath. I'm not sure why it isn't publshed in the plugin central yet.
- it's what allows using the Gradle Kotlin DSL in an idiomatic way.

For example, if you use `apply(plugin = ...)` in a Kotlin build script, You can't use the idiomatic

```
quarkus {
    // ...
}
```
 to configure the Quarkus extension. Instead, you need to add an import for the extension, and use something like

```
configure<QuarkusExtension> {
    // ...
}
```

Same for configuring the tasks added by the plugin: you don't benefit anymore for the type-safe, auto-completable extensions provided by Gradle when a plugin is applied by the plugins block.

**Comment by vorburger** [comment]
> avoid apply plugin xxx, and instead do as I suggest in quarkusio/quarkus#1403.

@jnizet I've done this in commit be5712a3537612d5e540eaf8dbdcd9a717924e65 on #90 - would you like to review?

> [quarkusio/quarkus#1323](https://github.com/quarkusio/quarkus/issues/1323) is now fixed, and when there is a new Quarkus 0.12.0 release I'll bump the versions in getting-started/build.gradle of [master...vorburger:gradle](https://github.com/quarkusio/quarkus-quickstarts/compare/master...vorburger:gradle) and raise a PR from that branch.

This happened in today's commit 9743fd7dbba43041598150b3b72dad24af4394b7 on #90 - now ready for merge, from my side.

**Comment by vorburger** [comment]
#90 was merged a while ago, so closing this now.  (But note new #136.)
